components/concept/engine/src/main/java/mozilla/components/concept/engine/Engine.kt

```diff
@@ -249,4 +249,13 @@ interface Engine : WebExtensionRuntime, DataCleanable {
      * Returns the version of the engine as [EngineVersion] object.
      */
     val version: EngineVersion
+
+    /**
+     * Add support for service workers requests to open new tabs.
+     *
+     * @param onAddTab - callback to adding a tab using the provided [EngineSession].
+     * Implementations are expected to add a new normal (not private) tab but not load any URL in that.
+     * The background service worker will use the same passed in [EngineSession] to load it's needed URL.
+     */
+    fun installServiceWorkersSupport(onAddTab: (engineSession: EngineSession) -> Unit) = Unit
```

Comment:
This breaks a bit with the current design. What do you think about adding a `registerServiceWorkerDelegate` function instead? Same as existing `registerWebNotificationDelegate`, `registerWebPushDelegate` etc.? The delegate would have a single `onOpenWindow` function.

Reply:
/cc @jonalmeida